internal/codespaces/api/api_test.go

```diff
@@ -270,13 +272,13 @@ func TestNew_ServerURL_customConfig(t *testing.T) {
 func TestNew_ServerURL_env(t *testing.T) {
 	t.Setenv("GITHUB_SERVER_URL", "https://mycompany.com")
 	t.Setenv("GITHUB_API_URL", "https://api.mycompany.com")
-	cfg := &config.ConfigMock{
-		AuthenticationFunc: func() *config.AuthConfig {
+	cfg := &ghmock.ConfigMock{
+		AuthenticationFunc: func() gh.AuthConfig {
 			return &config.AuthConfig{}
```

Comment:
Was the dropped pointer intentional?  Is it legal?

```suggestion
	cfg := &ghmock.ConfigMock{
		AuthenticationFunc: func() *gh.AuthConfig {
			return &config.AuthConfig{}
```

Reply:
`gh.AuthConfig` is an interface where as `config.AuthConfig` is a struct that was being constructed and `&` was taking the address.

Reply:
So to answer your question explicitly, yes it was intentional.